test/ndb_api_helpers_test.go

```diff
@@ -573,450 +505,3 @@ func TestResolveComputeProfileById_resolvesOk(t *testing.T) {
-
-	//Set
-	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		if !checkAuthTestHelper(r) {
-			t.Errorf("Invalid Authentication Credentials")
-		} else {
-			var response interface{}
-			if r.URL.Path == "/profiles" {
-				response = []ndb_api.ProfileResponse{
-					{
-						Id:              "1",
-						Name:            "a",
-						Type:            "test type",
-						EngineType:      "test engine",
-						LatestVersionId: "v-id-1",
-						Topology:        "test topology",
-						Status:          "READY",
-					},
-				}
-			} else if r.URL.Path == "/slas" {
-				response = []ndb_api.SLAResponse{
-					{
-						Id:                 "sla-1-id",
-						Name:               "SLA 1",
-						UniqueName:         "SLA 1 Unique Name",
-						Description:        "SLA 1 Description",
-						DailyRetention:     1,
-						WeeklyRetention:    2,
-						MonthlyRetention:   3,
-						QuarterlyRetention: 4,
-						YearlyRetention:    5,
-					},
-					{
-						Id:                 "sla-2-id",
-						Name:               "SLA 2",
-						UniqueName:         "SLA 2 Unique Name",
-						Description:        "SLA 2 Description",
-						DailyRetention:     1,
-						WeeklyRetention:    2,
-						MonthlyRetention:   3,
-						QuarterlyRetention: 4,
-						YearlyRetention:    5,
-					},
-					{
-						Id:                 NONE_SLA_ID,
-						Name:               common.SLA_NAME_NONE,
-						UniqueName:         "SLA 3 Unique Name",
-						Description:        "SLA 3 Description",
-						DailyRetention:     1,
-						WeeklyRetention:    2,
-						MonthlyRetention:   3,
-						QuarterlyRetention: 4,
-						YearlyRetention:    5,
-					},
-				}
-			}
-			resp, _ := json.Marshal(response)
-			w.WriteHeader(http.StatusOK)
-			w.Write(resp)
-		}
-	}))
-	defer server.Close()
-	ndb_client := ndb_client.NewNDBClient("username", "password", server.URL, "", true)
-
-	//Test
-	dbTypes := []string{common.DATABASE_TYPE_POSTGRES, common.DATABASE_TYPE_MYSQL, common.DATABASE_TYPE_MONGODB}
-	for _, dbType := range dbTypes {
-		dbSpec := v1alpha1.DatabaseSpec{
-			NDB: v1alpha1.NDB{
-				Server:           "abc.def.ghi.jkl/v99/api",
-				ClusterId:        "test-cluster-id",
-				CredentialSecret: "test-credential-secret-name",
-			},
-			Instance: v1alpha1.Instance{
-				DatabaseNames:        []string{"a", "b", "c", "d"},
-				Type:                 dbType,
-				DatabaseInstanceName: dbType + "-instance-test",
-				TimeZone:             "UTC",
-			},
-		}
-
-		reqData := map[string]interface{}{
-			common.NDB_PARAM_PASSWORD:       "qwertyuiop",
-			common.NDB_PARAM_SSH_PUBLIC_KEY: "",
-		}
-
-		db := &controller_adapters.Database{Database: v1alpha1.Database{
-			Spec: dbSpec,
-		}}
-
-		_, err := ndb_api.GenerateProvisioningRequest(context.Background(), ndb_client, db, reqData)
-		t.Log(err)
-		if err == nil {
-			t.Errorf("GenerateProvisioningRequest should return an error when ssh key is empty")
-		}
-	}
-}
```

Comment:
Refactored to use table test format and moved to `ndb_api/db_helpers_test.go` to be consistent with the convention set by @shenoypritika.